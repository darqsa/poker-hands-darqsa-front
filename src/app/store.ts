import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import handsSlice from "../features/hands/slices/handsSlice";
import uiSlice from "../features/ui/slices/uiSlice";
import userSlice from "../features/users/slices/userSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    hands: handsSlice,
    ui: uiSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
