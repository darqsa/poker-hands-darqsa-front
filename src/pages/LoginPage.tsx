import { Link } from "react-router-dom";
import Login from "../components/Login/Login";

const RegisterPage = (): JSX.Element => {
  return (
    <>
      <Login />
      <span>You have no account yet? Register </span>
      <Link to={`/register`}>
        <span>here</span>
      </Link>
    </>
  );
};
export default RegisterPage;
