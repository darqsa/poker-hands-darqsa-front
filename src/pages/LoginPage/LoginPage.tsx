import { Link } from "react-router-dom";
import Login from "../../components/Login/Login";
import ButtonStyled from "../../styles/ButtonStyled";
import LinkContainerStyled from "./LoginPageStyled";

const LoginPage = (): JSX.Element => {
  return (
    <>
      <Login />
      <LinkContainerStyled className="login-text-container">
        <span className="login-text-container__text">
          You have no account yet? Register{" "}
          <Link to={`/register`}>
            <ButtonStyled className="login-text-container__register-link">
              here
            </ButtonStyled>
          </Link>
        </span>
      </LinkContainerStyled>
    </>
  );
};
export default LoginPage;
