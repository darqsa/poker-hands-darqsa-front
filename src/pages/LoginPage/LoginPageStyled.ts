import styled from "styled-components";
import styles from "../../styles/styles";

const LinkContainerStyled = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  text-align: center;
  .login-text-container {
    &__text {
      color: ${styles.colors.greyText};
      font-size: ${styles.fontSizes.smaller};
      font-weight: ${styles.fontWeights.regular};
    }
    &__register-link {
      font-size: ${styles.fontSizes.smaller};
      font-weight: ${styles.fontWeights.regular};
    }
  }
`;
export default LinkContainerStyled;
