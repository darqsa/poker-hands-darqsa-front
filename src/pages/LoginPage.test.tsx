import { BrowserRouter } from "react-router-dom";
import LoginPage from "./LoginPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../app/store";

describe("Given a login page component", () => {
  describe("When invoked", () => {
    test("Then it should match the snapshot", () => {
      const expectedRegisterPage = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <LoginPage />
          </BrowserRouter>
        </Provider>
      );

      expect(expectedRegisterPage).toMatchSnapshot();
    });
  });
});
