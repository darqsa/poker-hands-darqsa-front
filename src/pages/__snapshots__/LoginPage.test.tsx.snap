// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a login page component When invoked Then it should match the snapshot 1`] = `
Array [
  <form
    className="sc-bczRLJ bgmofV form"
    onSubmit={[Function]}
  >
    <h2
      className="form__heading"
    >
      Create your account
    </h2>
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        autoComplete="off"
        className="form__input "
        id="username"
        onChange={[Function]}
        placeholder="Enter your username"
        required={true}
        value=""
      />
    </div>
    <div
      className="form__group"
    >
      <label
        className="form__label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="off"
        className="form__input "
        id="password"
        onChange={[Function]}
        placeholder="Enter your password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="form__button"
      disabled={true}
      type="submit"
    >
      Login
    </button>
  </form>,
  <span>
    You have no account yet? Register 
  </span>,
  <a
    href="/register"
    onClick={[Function]}
  >
    <span>
      here
    </span>
  </a>,
]
`;
